core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java

```diff
@@ -596,8 +591,13 @@ String getClientName(String account) throws BadRequestException {
     if (sep10Config.isClientAttributionRequired() && client == null) {
       throw new BadRequestException("Client not found");
     }
-    if (client != null && !sep10Config.getAllowedClientDomains().contains(client.getDomain())) {
-      client = null;
+
+    if (client != null && client.getDomains() != null) {
```

Comment:
Not really... if custodial client didn't specify any domain, the following check will return true and client will stay the same
```
boolean hasDomainIntersection =
          new HashSet<>(sep10Config.getAllowedClientDomains()).containsAll(client.getDomains());
```

Reply:
Custodial client never specify a domain, so client.getDomains is always null
For a non-custodial domain, I don't understand why we are using getAllowedClientDomains instead of getAllowedClientNames, this is very confusing.
If client name is allowed, it's client domains must be allowed. And if the name is not allowed, then the domains are not allowed as well. This relates to all of my comments.
IMO, if we know the client name, we should use the check if client name is allowed or not.
I assume it's just for backward compatibility, but I don't remember for sure. If it is, we should create a task to clean it up, and annotate org.stellar.anchor.platform.config.PropertySep10Config#clientAllowList as deprecated